packages/sqs/lib/sqs/AbstractSqsConsumer.ts

```diff
@@ -220,6 +223,13 @@ export abstract class AbstractSqsConsumer<
         }
         const { parsedMessage, originalMessage } = deserializedMessage.result
 
+        if (this.isPublisherDeduplicationEnabled(parsedMessage)) {
```

Comment:
Shouldn't it be:
```suggestion
        if (this.isConsumerDeduplicationEnabled(parsedMessage)) {
```
?

Reply:
Absolutely. I've fixed it and added sync checks to other methods related to dedup as part of this consumer.

It's weird that the tests were passing. I'll investigate why.

Reply:
@kjamrog I cannot figure out how the tests are working without the deduplication config. I have some guesses but haven't yet found an exact issue, but to me doesn't mean that there is something wrong with dedup, but rather test config/some bug in the handlerSpy(?)

If you take a look at the test `does not consume message with the same deduplication key twice`, it has 2 checks - the first one checks (using `waitForMessageWithId`) if the message was successfully consumed, the latter one checks (using `checkForMessage`) if the message was not consumed upon subsequent publishing. The tests are passing both with and without the dedup config. Here is the interesting thing. I've noticed that when I replace the second check (`checkForMessage`) with `waitForMessageWithId` then when deduplication is enabled the test fails as a result of a timeout - which is expected and confirms that deduplication works as a consumer cannot consume the message. If I disable deduplication, then test fails, because the message was eventually consumed (which is also expected because there is no dedup configured). So there are different behavior depending on whether `checkForMessage` or `waitForMessage` is used. I suspect that it might be something with the `checkForMessage` behavior, although implementation of this looks simple and I haven't yet figured out what might be an issue there. Is `checkForMessage` actually intended to be called as part of a consumer?

cc @kibertoad

Reply:
Could it be that you're not waiting long enough between the second `.publish()` and the `checkForMessage()` call?
The `checkForMessage()` only looks for messages that have already been processed, whereas the message you just published might not have been delivered/handled yet.
Try adding a 1-second delay after the second publish to test this.

To address the issue, we could introduce a new `MessageProcessingResult`, i.e. `rejected` or `duplicate`, to explicitly indicate when a message was rejected because it is a duplicate.
We can then update the expectations to ensure that the `processingResult` is explicitly set to `rejected` or `duplicate` when this occurs.

Reply:
@eduard-fa Thanks for a great suggestion. I've introduced a new status `duplicate` and adjusted consumers. Now tests work as expected (if dedup is disabled, then they simply fail).

I'll do the same for consumer to keep it consistent.